### DISCUSSION TITLE
Face detection

### DISCUSSION THREAD

**Discussion by barabba2005** [emgucv/emgucv#675] [Q&A]
Hi all

I have 10 photo faces and I put them in an Image array;

List<Image<Gray, Byte>> TrainedFaces = new List<Image<Gray, byte>>();
string[] files = Directory.GetFiles(@"D:\FaceImages\", "*.jpg", SearchOption.TopDirectoryOnly);
foreach (var file in files){ 
Image<Gray, byte> trainedImages = new Image<Gray, byte>(file);
TrainedFaces[ImagesCount] = (trainedImages);
}

I should find a face that matches with a particular photo;
private Image<Gray, byte> metal = new Image<Bgr, Byte>(@"C:\Users\HP-620\Pictures\testPicture.jpg");

I've triyed a lot of web example but the results are no good.

Please help me

**Comment by emgucv** [comment]
Please follow the Open CV face recognition tutorial here:
https://docs.opencv.org/3.4/da/d60/tutorial_face_main.html

Benchmarking face recognition data set can be found here:
https://docs.opencv.org/3.4/da/d60/tutorial_face_main.html#tutorial_face_facedb

Performance highly depends on the face recognition algorithm you choose. Proper facial feature alignment is critical.
